src/main/java/uk/gov/ons/ctp/response/collection/exercise/distribution/SampleUnitDistributor.java

```diff
@@ -191,27 +173,87 @@ private void distributeSampleUnits(
                 new PageRequest(0, appConfig.getSchedules().getDistributionScheduleRetrievalMax()));
 
     if (!CollectionUtils.isEmpty(sampleUnitGroups)) {
-      String sampleUnitGroupsPks =
+      List<Integer> sampleUnitGroupPKs =
           sampleUnitGroups
               .stream()
-              .map(group -> group.getSampleUnitGroupPK().toString())
-              .collect(Collectors.joining(","));
-      log.with("sample_unit_group_pks", sampleUnitGroupsPks)
-          .debug("DISTRIBUTION retrieved sampleUnitGroup PKs");
-      sampleDistributionListManager.saveList(
-          DISTRIBUTION_LIST_ID,
-          sampleUnitGroups
-              .stream()
-              .map(group -> group.getSampleUnitGroupPK())
-              .collect(Collectors.toList()),
-          true);
+              .map(ExerciseSampleUnitGroup::getSampleUnitGroupPK)
+              .collect(Collectors.toList());
+      log.debug(
+          "DISTRIBUTION retrieved sampleUnitGroups, sampleUnitGroupPKs: {}",
+          sampleUnitGroupPKs.stream().map(Object::toString));
+      sampleDistributionListManager.saveList(DISTRIBUTION_LIST_ID, sampleUnitGroupPKs, true);
```

Comment:
this is .a terrible way of doing distributed/clustered work

Reply:
I agree, it's confusing. I haven't changed anything to do with creating/locking/deleting the list here. Might be an improvement better saved for a separate PR?